src/linear_algebra/pi.lean

```diff
@@ -70,6 +72,17 @@ begin
   exact (mem_bot _).2 (funext $ assume i, h i)
 end
 
+/-- The `linear_map` version of `add_monoid_hom.single` and `pi.single`. -/
+@[simps]
+def single [decidable_eq ι] (i : ι) : φ i →ₗ[R] (Πi, φ i) :=
+{ to_fun := pi.single i,
+  map_smul' := λ r x, begin
```

Comment:
Since you simplify `linear_map.single i x` to `pi.single i x`, it makes sense to add a lemma `pi.single_smul` somewhere near the definition of `pi.semimodule`. BTW, `function.update_eq_iff` gives a slightly shorter proof.
Otherwise LGTM.
bors d+

Reply:
We don't have `pi.single_add` yet either, but I think we should - I'll leave adding all of those lemmas to a follow-up PR.

Reply:
But I will have a go at golfing this proof.

Reply:
I came up with
```lean
  map_smul' := λ r x, begin
    ext j,
    have := (function.apply_update (λ i, (•) r) 0 i x j).symm,
    simp only [pi.zero_apply, smul_zero] at this,
    exact this,
  end,
```
Do you have a better proof in mind?